builders/precompiles/xcm-Interface.md

```diff
@@ -0,0 +1,233 @@
+        - `feeAssetItem` (uint32) - the index of the asset that will be used to pay fees
+        - `weight` (Weight memory) - the maximum gas to use in the whole operation. Setting uint64::MAX to `refTime` acts in practice as *unlimited weight*
+
+    === "Example"
+        - `dest` - ["1",[]]
+        - `beneficiary` - [0, ["0x01f831d83025f527daeed39a644d64d335a4e627b5f4becc78fb67f05976889a0600"]]
+        - `assets` - [[[1, ["0x010000000000000000000000000000000000000800"]], 1000000000000000000]]
+        - `feeAssetItem` - 0
+        - `weight` - [9223372036854775807, 9223372036854775807]
+
+## Interact with the Solidity Interface {: #interact-with-the-solidity-interface }
+
+### Checking Prerequisites {: #checking-prerequisites }
+
+To follow along with this tutorial, you will need to have your preferred EVM wallet configured and an account funded with native tokens. You can add Moonbeam to MetaMask wallet following this guide: [Interacting with Moonbeam Using MetaMask](/tokens/connect/metamask/){target=\_blank}.
+
+### Remix Set Up {: #remix-set-up }
+
+You can interact with the XCM interface precompile using [Remix](https://remix.ethereum.org){target=\_blank}. To add the precompile to Remix, you will need to:
+
+1. Get a copy of [`XcmInterface.sol`](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}
+2. Paste the file contents into a Remix file named `XcmInterface.sol`
+
+### Compile the Contract {: #compile-the-contract }
+
+Next, you will need to compile the interface in Remix:
+
+1. Click on the **Compile** tab, second from top
+2. Compile the interface by clicking on **Compile XcmInterface.sol**
+
+![Compiling XcmInterface.sol](/images/builders/precompiles/xcm-interface/xcm-interface-1.webp)
+
+When the compilation is completed, you will see a green checkmark next to the **Compile** tab.
+
+### Access the Contract {: #access-the-contract }
+
+Instead of deploying the precompile, you will access the interface given the address of the precompiled contract:
+
+1. Click on the **Deploy and Run** tab directly below the **Compile** tab in Remix. Please note that the precompiled contracts are already accessible at their respective addresses. Therefore, there is no deployment step
+2. Make sure **Injected Provider - Metamask** is selected in the **ENVIRONMENT** dropdown. Once you select **Injected Provider - Metamask**, you may be prompted by MetaMask to connect your account to Remix if it's not already connected
+3. Make sure the correct account is displayed under **ACCOUNT**
+4. Ensure **XCM - XcmInterface.sol** is selected in the **CONTRACT** dropdown. Given that it is a precompiled contract, there is no deployment step. Instead, you are going to provide the address of the precompile in the **At Address** field
+5. Provide the address of the precompile: `{{networks.moonbeam.precompiles.xcmInterface}}` and click **At Address**
```

Comment:
Currently showing as blank for me
<img width="567" alt="Screenshot 2024-08-07 at 10 37 57 AM" src="https://github.com/user-attachments/assets/b89c857a-d71a-47a7-bdf9-7ab946e69b9f">